### ISSUE TITLE
Future-proofing the example code

### ISSUE THREAD

**Issue by ISSOtm** [ISSOtm/gb-asm-tutorial#17]
Parts II and III will have the reader (and thus, the writer as well) working iteratively on a codebase. Where it gets iffy is that any change made to earlier lessons must be consistently reflected in later lessons. (This could happen for example because an earlier lesson is amended to make "room for" a later one, to update the syntax for a newer RGBDS version etc.)

Propagating these changes throughout the tutorial, where they would be presented "inline", would be a massive pain. Instead, I suggest the codebase should be checked out into Git, and inline code examples pulled from that repo.

This does get somewhat involved in several ways (e.g. ensuring that all referenced commits are reachable from the `master` branch, keeping referenced line numbers consistent when rebasing, etc.), but I think is essential if we want to avoid the tutorial irremediably bit-rotting in the longer term.

**Comment by avivace** [comment]
Agree, a simple CI/CD set up could also check the code examples against the target versions of RGBDS
